### PR TITLE
Update documentation

### PR DESCRIPTION
**Pull request by RoSk0**
## Problem/motivation

At the moment **Creating cores** section is hard to follow, doesn't list all supported options and doesn't have useful examples. Instead it send user to an example project which is outdated and also not easy to follow.

In my opinion it's always a good idea to keep documentation close to the code so there is less chance that it will become outdated.

## Proposed changes

- update documentation links to point to the new <https://solr.apache.org> project website
- update documentation links to point to new version `7.7` => `8.9`
- add sub-headers to  **Creating cores** section 
- add example of how to use custom configset
- add note about configset directory layout (personally was tripped by this multiple times)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -104,21 +106,34 @@ $ docker exec -it my_solr solr create_core -c gettingstarted
...
+
+### Using solr-create command
 
 The third option is to use the `solr-create` command. This runs a Solr in the background in the container, then uses the Solr control script to create the core, then stops the Solr server and restarts it in the foreground. This method is less popular because the double Solr run can be confusing.
 
 ```console
 $ docker run -d -p 8983:8983 --name my_solr solr:8 solr-create -c gettingstarted
 ```
 
+### Custom set up scripts
```
Comment (dsmiley) [inline README.md:135]:
```suggestion
### Custom set-up scripts
```

**Review by dsmiley — APPROVED** [review]
Awesome -- thanks for contributing!
I'll give another day or two for further feedback but looks good to me.

**Comment by ctargett** [comment]
I thought this documentation migrated to the Solr Ref Guide with https://issues.apache.org/jira/browse/SOLR-14790 and apache/solr#101? At a minimum shouldn't parallel changes be made to those pages?

**Comment by dsmiley** [comment]
Oh, I forgot the docs here have been migrated; thanks for mentioning it.  Yeah, parallel changes should be made.  At least these don't get touched often (for better/worse).
